packages/medusa/src/services/order.ts

```diff
@@ -1144,13 +1144,26 @@ class OrderService extends TransactionBaseService {
 
       const inventoryServiceTx =
         this.productVariantInventoryService_.withTransaction(manager)
+
+      const previouslyFulfilledQuantities = order.fulfillments.reduce(
+        (acc, f) => {
+          return f.items.reduce((acc, item) => {
+            return {
```

Comment:
suggestion(non blocking):

```ts
acc[item.item_id] = (acc[item.item_id] || 0) + item.quantity
return acc
```

we don't really need to copy the entire object all the time